code/modules/surgery/bodyparts/dismemberment.dm

```diff
@@ -293,36 +299,36 @@
 			pill.forceMove(src)
 
 	//Make sure de-zombification happens before organ removal instead of during it
-	var/obj/item/organ/zombie_infection/ooze = owner.getorganslot(ORGAN_SLOT_ZOMBIE)
-	if(istype(ooze))
-		ooze.transfer_to_limb(src, owner)
-
-	name = "[owner.real_name]'s head"
+	name = owner ? "[owner.real_name]'s head" : "unknown [limb_id] head" //PARIAH STATION EDIT
 	..()
 
 //Attach a limb to a human and drop any existing limb of that type.
-/obj/item/bodypart/proc/replace_limb(mob/living/carbon/limb_owner, special)
+/obj/item/bodypart/proc/replace_limb(mob/living/carbon/C, special, is_creating = FALSE) //PARIAH STATION EDIT
 	if(!istype(limb_owner))
 		return
-	var/obj/item/bodypart/limb = limb_owner.get_bodypart(body_zone) //needs to happen before attach because multiple limbs in same zone breaks helpers
-	if(!attach_limb(limb_owner, special))//we can attach this limb and drop the old after because of our robust bodyparts system. you know, just for a sec.
-		return
+	var/obj/item/bodypart/O = C.get_bodypart(body_zone) //PARIAH STATION EDIT
 	if(limb)
 		limb.drop_limb(1)
-
+	return attach_limb(C, special, is_creating) //PARIAH STATION EDIT
 /obj/item/bodypart/head/replace_limb(mob/living/carbon/head_owner, special)
 	if(!istype(head_owner))
 		return
 	var/obj/item/bodypart/head/head = head_owner.get_bodypart(body_zone)
-	if(!attach_limb(head_owner, special))
-		return
 	if(head)
-		head.drop_limb(1)
+		if(!special) //PARIAH STATION EDIT START
+			return
+		else
+			O.drop_limb(1)
+	return attach_limb(C, special, is_creating)
+
+/obj/item/bodypart/proc/attach_limb(mob/living/carbon/C, special, is_creating = FALSE)
+	var/obj/item/bodypart/chest/mob_chest = C.get_bodypart(BODY_ZONE_CHEST)
+	if(mob_chest && !(mob_chest.acceptable_bodytype & bodytype)) //PARIAH STATION EDIT END
 
 /obj/item/bodypart/proc/attach_limb(mob/living/carbon/new_limb_owner, special)
 	if(SEND_SIGNAL(new_limb_owner, COMSIG_CARBON_ATTACH_LIMB, src, special) & COMPONENT_NO_ATTACH)
 		return FALSE
```

Comment:
You only want one of these. :P
https://github.com/shiptest-ss13/Shiptest/blob/b6c422ae7864cc5f2c9f60aecb483f457e902782/code/modules/surgery/bodyparts/dismemberment.dm#L282-L285
```suggestion
/obj/item/bodypart/proc/attach_limb(mob/living/carbon/new_limb_owner, special, is_creating = FALSE)
	var/obj/item/bodypart/chest/mob_chest = new_limb_owner.get_bodypart(BODY_ZONE_CHEST)
	if(mob_chest && !(mob_chest.acceptable_bodytype & bodytype)) //PARIAH STATION EDIT END
		return FALSE
```